### PR TITLE
Add while loop to check for more busy ports

### PR DESCRIPTION
**Pull request by leonardovida**
The current code ` if not input_args.debug and check_port_in_use('localhost', port):` is checking whether the port (e.g. default :5000) is in use and then adds +1, but checks only once. If a user has a range of ports in use, for example 5000-5002, `label studio` does not manage to find an open port.

Modified by moving `check_port_in_use()` into a while loop to continue the check until there is a free port `label studio` can use.

I also took the freedom to create a PR without opening an issue first, hope that is fine with you.

### CODE REVIEW

File: label_studio/blueprint.py [hunk 1]
```diff
@@ -981,12 +981,13 @@ def main():
...
-            print('\n*** WARNING! ***\n* Port ' + str(old_port) + ' is in use.\n' +
-                  '* Trying to start at ' + str(port) +
-                  '\n****************\n')
+        if not input_args.debug:
+            while check_port_in_use('localhost', port):
+                old_port = port
+                port = int(port) + 1
+                print('\n*** WARNING! ***\n* Port ' + str(old_port) + ' is in use.\n' +
+                    '* Trying to start at ' + str(port) +
+                    '\n****************\n')
```
Comment (makseq) [inline label_studio/blueprint.py:990]: Could you add here a breaking condition to avoid endless loop in the edge case?

Comment (makseq) [reply label_studio/blueprint.py:990]: @leonardovida

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio/blueprint.py (hunk 1 above)
Comment (leonardovida) [reply label_studio/blueprint.py:990]: @makseq Sure, but if you mean when port > 65535, `socket.connect_ex()` already takes care of that raising an `OverflowError`. If you mean something else, could you please specify?

Comment (makseq) [reply label_studio/blueprint.py:990]: Maybe we can limit the range to +1000 ports? I think it could take a lot of time until 65535...

Comment (makseq) [reply label_studio/blueprint.py:990]: @leonardovida I think we must be hurry to be included into 0.9.1 release :smiley:

Comment (leonardovida) [reply label_studio/blueprint.py:997]: @makseq Makes super sense limiting to +1000. I'm raising a `ConnectionError()`. Let me know if you want a more informative message or just want to `break` the loop and continue execution!

Comment (leonardovida) [reply label_studio/blueprint.py:997]: Forgot the lack of support for fStrings. Sorry for the usage of actions

**Pull request merged**